rcongui/src/components/PlayerInfo/PlayerInfo.js

```diff
@@ -0,0 +1,267 @@
+        location: PropTypes.object.isRequired,
+        history: PropTypes.object.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            steam_id_64: "",
+            created: "0",
+            names: [],
+            sessions: [],
+            sessions_count: 1086,
+            total_playtime_seconds: 4495950,
+            current_playtime_seconds: 17455,
+            received_actions: [],
+            penalty_count: {
+                TEMPBAN: 0,
+                PERMABAN: 0,
+                PUNISH: 0,
+                KICK: 0
+            },
+            blacklist: {
+                is_blacklisted: false,
+                reason: "",
+                by: ""
+            },
+            flags: [],
+            watchlist: {},
+            steaminfo: {},
+            perma: false,
+            temp: false,
+            vip: false,
+            loaded: false
+        }
+    }
+
+    /**
+     * fetch bans that currently affect the steamId64
+     * @param steamId64
+     */
+    #fetchPlayerBan = (steamId64) => {
+        get(`get_ban?steam_id_64=${steamId64}`)
+            .then((data) => {
+                const temp = data.result.find((ban, index) => {
+                    if (ban.type === "temp")
+                        return true
+                })
+                const perma = data.result.find((ban, index) => {
+                    if (ban.type === "perma")
+                        return true
+                })
+                if (temp !== undefined) {
+                    this.setState({temp: true})
+                }
+                if (perma !== undefined) {
+                    this.setState({perma: true})
+                }
+            })
+            .catch(handle_http_errors);
+    }
+
+    /**
+     * fetch Player data
+     * @param steamId64
+     */
+    #fetchPlayer = (steamId64) => {
+        get(`player?steam_id_64=${steamId64}`)
+            .then((response) => showResponse(response, "get_user", false))
+            .then((data) => {
+                this.setState({
+                    created: data.result.created,
+                    names: data.result.names,
+                    sessions: data.result.sessions,
+                    sessions_count: data.result.sessions_count,
+                    total_playtime_seconds: data.result.total_playtime_seconds,
+                    current_playtime_seconds: data.result.current_playtime_seconds,
+                    received_actions: data.result.received_actions,
+                    penalty_count: data.result.penalty_count,
+                    blacklist: data.result.blacklist,
+                    flags: data.result.flags,
+                    watchlist: data.result.watchlist,
+                    steaminfo: data.result.steaminfo,
+                    loaded: true
+                })
+
+            })
+            .catch(handle_http_errors);
+    }
+
+    componentWillReceiveProps(nextProps){
```

Comment:
Why do you do it here and in the componentDidMount?

I think this is unecessary, if the propery changes it should trigger a re-render anyways